internal/builtinfunctions/functions.go

```diff
@@ -164,7 +166,46 @@ func getFloatToStringFunction() schema.CallableFunction {
 	return funcSchema
 }
 
-// TODO: Webb can add a function called floatToStringAdvanced that allows other float formats
+func getFloatToFormattedStringFunction() schema.CallableFunction {
+	funcSchema, err := schema.NewCallableFunction(
+		"floatToFormattedString",
+		[]schema.Type{
+			schema.NewFloatSchema(nil, nil, nil),
+			schema.NewStringSchema(nil, nil, regexp.MustCompile(`^[beEfgGxX]$`)),
+			schema.NewIntSchema(nil, nil, nil),
+		},
+		// 'b' format: -ddddp±ddd
+		// 'e' format: -d.dddde±dd
+		// 'E' format: -d.ddddE±dd
+		// 'f' format: -ddd.dddd
+		// 'x' format: -0xd.ddddp±ddd
+		// 'X' format: -0Xd.ddddP±ddd
+		schema.NewStringSchema(
+			nil,
+			nil,
+			regexp.MustCompile(`^-?(?:0[xX])?\d+(?:\.\d*)?(?:[pPeE][-+]\d{2,3})?$`)),
+		false,
+		schema.NewDisplayValue(
+			schema.PointerTo("floatToFormattedString"),
+			schema.PointerTo(
+				"Converts a floating point number to a string according to the "+
+					"specified formatting directive and precision."+
+					" Param 1: the floating point value to convert\n"+
+					" Param 2: the format specifier: 'b', 'e', 'E', 'f', 'g', 'G', 'x', 'X'\n"+
+					" Param 3: the number of digits included in the fraction portion.\n"+
```

Comment:
The `-1` option is not very obvious in Go's documentation, so it may make sense to mention that specific special case in this description.